control-plane-operator/controllers/hostedcontrolplane/cno/clusternetworkoperator.go

```diff
@@ -0,0 +1,251 @@
+			KubeRBACProxy:                images["kube-rbac-proxy"],
+			Multus:                       images["multus-cni"],
+			MultusAdmissionController:    images["multus-admission-controller"],
+			CNIPlugins:                   images["container-networking-plugins"],
+			BondCNIPlugin:                images["network-interface-bond-cni"],
+			WhereaboutsCNI:               images["multus-whereabouts-ipam-cni"],
+			RouteOverrideCNI:             images["multus-route-override-cni"],
+			MultusNetworkPolicy:          images["multus-networkpolicy"],
+			OVN:                          images["ovn-kubernetes"],
+			EgressRouterCNI:              images["egress-router-cni"],
+			KuryrDaemon:                  images["kuryr-cni"],
+			KuryrController:              images["kuryr-controller"],
+			NetworkMetricsDaemon:         images["network-metrics-daemon"],
+			NetworkCheckSource:           images["cluster-network-operator"],
+			NetworkCheckTarget:           images["cluster-network-operator"],
+			CloudNetworkConfigController: images["cloud-network-config-controller"],
+			TokenMinter:                  images["token-minter"],
+		},
+		ReleaseVersion:          version,
+		AvailabilityProberImage: images[util.AvailabilityProberImageName],
+		OwnerRef:                config.OwnerRefFrom(hcp),
+	}
+
+	p.DeploymentConfig.Scheduling.PriorityClass = config.DefaultPriorityClass
+	p.DeploymentConfig.SetColocation(hcp)
+	p.DeploymentConfig.SetRestartAnnotation(hcp.ObjectMeta)
+	p.DeploymentConfig.SetReleaseImageAnnotation(hcp.Spec.ReleaseImage)
+	p.DeploymentConfig.SetControlPlaneIsolation(hcp)
+	p.DeploymentConfig.Replicas = 1
+	p.DeploymentConfig.SetDefaultSecurityContext = setDefaultSecurityContext
+	p.APIServerAddress = hcp.Status.ControlPlaneEndpoint.Host
+	p.APIServerPort = hcp.Status.ControlPlaneEndpoint.Port
+	p.HostedClusterName = hcp.Name
+	p.TokenAudience = hcp.Spec.IssuerURL
+
+	return p
+}
+
+func ReconcileServiceAccount(sa *corev1.ServiceAccount, ownerRef config.OwnerRef) error {
+	ownerRef.ApplyTo(sa)
+	return nil
+}
+
+func ReconcileRole(role *rbacv1.Role, ownerRef config.OwnerRef) error {
+	ownerRef.ApplyTo(role)
+	// Required by CNO to manage ovn-kubernetes control plane components
+	role.Rules = []rbacv1.PolicyRule{
+		{
+			APIGroups: []string{corev1.SchemeGroupVersion.Group},
+			Resources: []string{
+				"events",
+				"configmaps",
+				"pods",
+				"secrets",
+				"services",
+			},
+			Verbs: []string{"*"},
+		},
+		{
+			APIGroups: []string{"policy"},
+			Resources: []string{"poddisruptionbudgets"},
+			Verbs:     []string{"*"},
+		},
+		{
+			APIGroups: []string{appsv1.SchemeGroupVersion.Group},
+			Resources: []string{"statefulsets"},
+			Verbs:     []string{"*"},
+		},
+		{
+			APIGroups: []string{routev1.SchemeGroupVersion.Group},
+			Resources: []string{"routes"},
+			Verbs:     []string{"*"},
+		},
+		{
+			APIGroups: []string{hyperv1.GroupVersion.Group},
+			Resources: []string{
+				"hostedcontrolplanes",
```

Comment:
@kyrtapz  minor thing, we'd need `"hostedcontrolplanes/status"` resource and `update` verb later for hostedcontrolplane update in status manager PR in CNO.

Reply:
done